account_journal_general_sequence/tests/test_numbering.py

```diff
@@ -85,3 +85,20 @@ def test_install_no_entry_number(self):
         invoice.action_post()
         # Ensure there's no entry number
         self.assertFalse(invoice.entry_number)
+
+    @users("admin")
+    def test_new_company_journal(self):
+        cmp = self.env["res.company"].create({"name": "Company100", "currency_id": 2})
+        self.env["account.journal"].create(
+            {
+                "name": "Journal_test100",
+                "code": "code100",
+                "type": "bank",
+                "company_id": cmp.id,
+            }
+        )
+        self.env.company_id = cmp.id
+        new_company_invoice = self._create_invoice(
+            date_invoice="2023-07-01", auto_validate=True
+        )
+        self.assertEqual(new_company_invoice.entry_number, "2023/000001")
```

Comment:
You'll have to change the assertion once you revert the change on the padding number:

```suggestion
        self.assertEqual(new_company_invoice.entry_number, "2023/0000000001")
```